### DISCUSSION TITLE
Log to file

### DISCUSSION THREAD

**Discussion by MoralCode** [UmerCodez/SensorServer#30] [Ideas]
Seems like being able to write data out to files would enable a lot more use cases, such as Matt did in this video: https://www.youtube.com/watch?v=sB2X5l5CsNs

**Comment by UmerCodez** [comment]
@MoralCode You can still use this app to log sensor values to a file. Here's a simple way to do it: Install [Pydroid 3](https://play.google.com/store/apps/details?id=ru.iiec.pydroid3&hl=en&gl=US&pli=1) on your Android device, and it will enable you to run Python scripts. In the onMessage() method, write the sensor values to a file. This file will be saved in your phone's internal storage.

 ```python
...
log = open("data.txt","w")
def on_message(ws, message):
    values = json.loads(message)['values']
    x = values[0]
    y = values[1]
    z = values[2]
    log.write(f"x = {x} , y = {y} , z = {z}")

...

```
